aws/resource_aws_networkfirewall_firewall.go

```diff
@@ -50,6 +50,42 @@ func resourceAwsNetworkFirewallFirewall() *schema.Resource {
 				Type:     schema.TypeBool,
 				Optional: true,
 			},
+			"firewall_status": {
```

Comment:
schema here emulates the API with the exception that SyncStates can't be represented with a nested map so a TypeList is used in place (ref: https://docs.aws.amazon.com/network-firewall/latest/APIReference/API_FirewallStatus.html)

seems a bit nested just to retrieve the `endpoint_id` so maybe worth straying away from the API in this case?